src/perf/shader.md

```diff
@@ -1,33 +1,45 @@
-shader captured in the Skia Shader Language (SkSL) for a
-significant improvement.
-
-![Side-by-side screenshots of janky mobile app next to non-janky app]({{site.url}}/assets/images/docs/perf/render/shader-jank.gif)
+If the animations on your mobile app appear to be janky, but only on the first
+run, this is likely due to shader compilation. Flutter's long term solution to
+shader compilation jank is [Impeller][], which is in developer preview on master
+channel for iOS. Before continuing with the instructions below, please try out
+Impeller on iOS, and let us know in a GitHub issue if it doesn't address your
+issue.
+
+While we work on making Impeller production ready, it is possible to mitigate
+shader compilation jank by bundling precompiled shaders with an iOS app.
+Unfortunately, this approach does not work well on Android due to precompiled
+shaders being device or GPU specific. The Android hardware ecosystem is diverse
+enough that the GPU specific precompiled shaders bundled with an application
+will work on only a small subset of devices, and will likely make jank worse on
+the other devices, or even create rendering errors.
+
+Also please note that we are not planning to make improvements to the developer
+experience for creating precompiled shaders described below. Instead, we are
+focusing our energies on the more robust solution to this problem that Impeller
+offers.
 
 ## What is shader compilation jank?
 
-If an app has janky animations during the first run,
-and later becomes smooth for the same animation,
-then it's very likely due to shader compilation jank.
+A shader is a piece of code that runs on a GPU (graphics processing unit). When
+the Skia graphics backend that Flutter uses for rendering sees a new sequence
+of draw commands for the first time, it will sometimes generate and compile a
+custom GPU shader for that sequence of commands. This allows that sequence and
+potentially similar sequences to render as fast as possible.
 
-More technically, a shader is a piece of code that runs on
-a GPU (graphics processing unit).
-When a shader is first used, it needs to be compiled on the device.
-The compilation could cost up to a few hundred milliseconds
-whereas a smooth frame needs to be drawn within 16 milliseconds
-for a 60 fps (frame-per-second) display.
-Therefore, a compilation could cause tens of frames to be missed,
-and drop the fps from 60 to 6. This is _compilation jank_.
-After the compilation is complete, the animation should be smooth.
+Unfortunately, Skia's shader generation and compilation happen in sequence with
+the frame workload. The compilation could cost up to a few hundred milliseconds
+whereas a smooth frame needs to be drawn within 16 milliseconds for a 60 fps
+(frame-per-second) display. Therefore, a compilation could cause tens of frames
+to be missed, and drop the fps from 60 to 6. This is _compilation jank_. After
+the compilation is complete, the animation should be smooth.
```

Comment:
Consider adding some information here about how impeller will do this differently?